### DISCUSSION TITLE
Mapsforge rendering layers

### DISCUSSION THREAD

**Discussion by AFA141** [mapsforge/mapsforge#1398] [Q&A] [answered]
Hello, 

I have two questions.

Does mapsforge supports more rendering layers? For example in https://www.openstreetmap.org, we can choose more layers like CyclOSM, TransportMap etc..
And is there any way to show contour lines on map?

I am using library on Desktop.

**Comment by devemux86** [comment, marked as answer]
These are bitmap tile maps.

> Does mapsforge supports more rendering layers?

You can create custom tile sources, see the [DownloadCustomLayerViewer](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/DownloadCustomLayerViewer.java) example.

> And is there any way to show contour lines on map?

Contour lines can be rendered if they exist in the map file and render theme.
e.g. [OpenAndroMaps](https://www.openandromaps.org/en) maps and [Elevate](https://www.openandromaps.org/en/legend/elevate-mountain-hike-theme) theme
